src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -672,33 +672,47 @@ private TypeSymbol GetTypeOfTypeDef(TypeDefinitionHandle typeDef, out bool isNoP
 
         /// <exception cref="UnsupportedSignatureContent">If the encoded type is invalid.</exception>
         /// <exception cref="BadImageFormatException">An exception from metadata reader.</exception>
-        private ImmutableArray<ModifierInfo<TypeSymbol>> DecodeModifiersOrThrow(ref BlobReader signatureReader, out SignatureTypeCode typeCode)
+        private ImmutableArray<ModifierInfo<TypeSymbol>> DecodeModifiersOrThrow(
+            ref BlobReader signatureReader,
+            out SignatureTypeCode typeCode,
+            Func<TypeSymbol, bool> acceptRequiredModifier = null)
```

Comment:
>Func<TypeSymbol, bool> acceptRequiredModifier = null [](start = 12, length = 52)

This feels too complicated. I think we can simply pas a boolean whether IsConst modifier should be accepted as a mod_req or not.